server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -100,6 +114,102 @@ export class MetadataExtractionProcessor {
     }
   }
 
+  private async extractPixelLivePhoto(
+    asset: AssetEntity,
+    directory: DirectoryEntry[],
+    fileCreatedAt: Date | null,
+  ): Promise<AssetEntity | null> {
+    if (asset.livePhotoVideoId) {
+      // Already extracted, don't try again.
+      const [ret] = await this.assetRepository.getByIds([asset.livePhotoVideoId]);
+      this.logger.log(`Already extracted asset ${ret.originalPath}.`);
+      return ret;
+    }
+    let foundMotionPhoto = false;
+    let motionPhotoOffsetFromEnd = 0;
+    let motionPhotoLength = 0;
+    let motionPhotoMime = '';
+
+    // Look for the directory entry with semantic label "MotionPhoto", which is the embedded video.
+    // Then, determine the length from the end of the file to the start of the embedded video.
+    for (let i = 0; i < directory.length; i++) {
+      if (directory[i].Item.Semantic == 'MotionPhoto') {
+        if (foundMotionPhoto) {
+          this.logger.error(`Asset ${asset.originalPath} has more than one motion photo.`);
+          return null;
```

Comment:
This should probably be continue not return

Reply:
Done.